### ISSUE TITLE
I want to display the message "Verification is not complete.

### ISSUE THREAD

**Issue by ngadceser** [better-auth/better-auth#1050]
### Is this suited for github?

- [ ] Yes, this is suited for github

### To Reproduce

Hello, This is part of the example that I wrote.
---
export const auth = betterAuth({
  database: prismaAdapter(prisma, {
    provider: 'mongodb',
  }),
  plugins: [nextCookies(), openAPI()],
  emailAndPassword: {
    enabled: true,
    requireEmailVerification: true,
  },
---
const response = await auth.api.signInEmail({
    body: {
        email: "",
        password: ""
    },
    asResponse: true
})
---
For users who repeatedly attempt  to sign-in without clicking the verification link 
after sign-up, 
the current behavior is to resend the verification email infinitely 
every time a login attempt is made.

The functionality I want to implement is to simply display a message 
like "Verification is not complete" instead of automatically resending the email.  

const response = wait auth.api.signInEmail() 
There is also a way to check in DB before doing it, but by any chance

Is there an option I might not be aware of to achieve this?  
Thank you in advance!

### Current vs. Expected behavior

I just want to show you the message, "The verification is not complete."

### What version of Better Auth are you using?

1.1.3

### Provide environment information

```bash
- OS: Windows11
-
```

### Which area(s) are affected? (Select all that apply)

Other

### Auth config (if applicable)

```typescript
export const auth = betterAuth({
  database: prismaAdapter(prisma, {
    provider: 'mongodb',
  }),
  plugins: [nextCookies(), openAPI()],
  emailAndPassword: {
    enabled: true,
    requireEmailVerification: true,
  },
  emailVerification: {
    sendOnSignUp: true,
    autoSignInAfterVerification: true,
    sendVerificationEmail: async ({ user, token }) => {
      const verificationUrl = `${process.env.BETTER_AUTH_URL}/api/auth/verify-email?token=${token}&callbackURL=${process.env.EMAIL_VERIFICATION_CALLBACK_URL}`;
      // Send the verification email here
      await mailer.sendMail({
        from: process.env.EMAIL_FROM ?? 'admin@localhost',
        to: user.email,
        subject: 'Verify your email address',
        html: `<p>Click the following link to verify your email address:</p>
                 <p><a href="${verificationUrl}">Verify Link</a></p>`,
      });
    },
  },
} satisfies BetterAuthOptions);
export type Session = typeof auth.$Infer.Session;
```

### Additional context

...

**Comment by daveycodez** [comment]
+1 `emailVerification.sendOnSignIn` would be a good solution here

**Comment by ngadceser** [comment]
> `emailVerification.sendOnSignIn`

I would love it if there was a property to do "sendOnSignIn: false".

**Comment by Nicolab** [comment]
Yes, it will be more clean. To keep the link with a related issue https://github.com/better-auth/better-auth/issues/788

**Comment by Nicolab** [comment]
> Hi, [@ngadceser](https://github.com/ngadceser). I'm [Dosu](https://dosu.dev), and I'm helping the better-auth team manage their backlog. I'm marking this issue as stale.
> 
> **Issue Summary**
> 
> * You requested help to modify the authentication system to display a message instead of resending a verification email on sign-in attempts.
> * User daveycodez suggested using a property like `emailVerification.sendOnSignIn` to control this behavior.
> * Nicolab linked a related issue for further context, and you expressed interest in disabling the email resend feature.
> * The suggestion received positive feedback and seems to meet your needs.
> 
> **Next Steps**
> 
> * Is this issue still relevant to the latest version of the better-auth repository? If so, please comment to keep the discussion open.
> * Otherwise, this issue will be automatically closed in 7 days.
> 
> Thank you for your understanding and contribution!

Yes this issue is still relevant to the latest version.

**Comment by frectonz** [comment]
`sendOnSignIn` is already available
